### ISSUE TITLE
Comparing h5coro to referenceFileSystem approach

### ISSUE THREAD

**Issue by rsignell-usgs** [ICESat2-SlideRule/sliderule-icesat2#3]
I just found out about the h5coro effort, and I would be interesting to compare the performance to the approach we have been using for [cloud-performant reading of HDF5](https://medium.com/pangeo/cloud-performant-netcdf4-hdf5-with-zarr-fsspec-and-intake-3d3a3e7cb935), where we read HDF5 format using the Zarr library. 

Where might I find the h5coro [performance tests](http://icesat2sliderule.org/h5coro/#performance-comparisons) so I could try these using the Zarr approach?

**Comment by jpswinski** [comment]
@rsignell-usgs I read the article you linked, and the work you are doing looks very exciting, especially since your solution is a pure Python one.  I've done my best to answer your question below, but if you have any follow up questions, or want to connect, feel free to send me an email using the email address I have listed in my GitHub profile.

The performance tests we ran to compare HSDS and H5Coro are going to be very hard to recreate outside our environment, and were really only intended to compare the two in relative terms, not provide an absolute performance estimate.  

But if you wanted to go down the path of trying to recreate the test using Zarr, here are the basic elements of the test:

**Region of Interest**
{"lon": -108.3435200747503, "lat": 38.89102961045247},             
{"lon": -107.7677425431139, "lat": 38.90611184543033},              
{"lon": -107.7818591266989, "lat": 39.26613714985466},             
{"lon": -108.3605610678553, "lat": 39.25086131372244},             
{"lon": -108.3435200747503, "lat": 38.89102961045247}

**Source Data**
ICESat-2/ATL03 (https://nsidc.org/data/ATL03)

**Full Dataset Reads**
geolocation/reference_photon_lat
geolocation/reference_photon_lon
geolocation/segment_ph_cnt

**Subsetted Data Set Reads**
ancillary_data/atlas_sdp_gps_epoch
orbit_info/sc_orient
ancillary_data/start_rgt
ancillary_data/end_rgt
ancillary_data/start_cycle
ancillary_data/end_cycle
geolocation/delta_time
geolocation/segment_id
geolocation/segment_dist_x
heights/dist_ph_along
heights/h_ph
heights/signal_conf_ph
bckgrd_atlas/delta_time
bckgrd_atlas/bckgrd_rate

**Processing**
Each read (both full and subsetted) is performed twice - one time that focusses on processing what is read and returning a small gridded elevation set, and one time that focusses on just reading the data and returning it.  The combined time is what was used in the test. Also, NASA's CMR system will return many more granules for the region of interest than actually intersect, due to default off-pointing margins added to all of the ground tracks.  This is okay, and was included in the test because the purpose of the test was to capture as realistic as possible processing runs against ICESat-2 data.
